packages/web-frontend/src/containers/Map/Map.js

```diff
@@ -188,38 +133,11 @@ class MapComponent extends Component {
         <TileLayer tileSetUrl={tileSetUrl} />
         <ZoomControl sidePanelWidth={sidePanelWidth} />
         <DemoLand />
-        {currentOrganisationUnit && organisationUnitIsArea(currentOrganisationUnit) && (
-          <InteractivePolygon
-            area={currentOrganisationUnit}
-            organisationUnitChildren={getChildren(currentOrganisationUnit.organisationUnitCode)}
-            isActive
-            {...basicPropsForInteractivePolygon}
-          />
-        )}
-        {displayedChildren?.map(area => (
-          <InteractivePolygon
-            area={area}
-            key={area.organisationUnitCode}
-            organisationUnitChildren={getChildren(area.organisationUnitCode)}
-            isChildArea
-            {...basicPropsForInteractivePolygon}
-          />
+        {displayedMapOverlayCodes.map(displayedMapOverlayCode => (
+          <DataVisualsLayer displayedMapOverlayCode={displayedMapOverlayCode} />
```

Comment:
Instead of selecting/combining two sets of measure data and series, here we manage to handle it separately.